lib/internal/errors.js

```diff
@@ -479,11 +480,16 @@ function makeNodeErrorWithCode(Base, key) {
  * @returns {T}
  */
 function hideStackFrames(fn) {
-  // We rename the functions that will be hidden to cut off the stacktrace
-  // at the outermost one
-  const hidden = nodeInternalPrefix + fn.name;
-  ObjectDefineProperty(fn, 'name', { __proto__: null, value: hidden });
-  return fn;
+  function wrappedFn(...args) {
+    try {
+      return ReflectApply(fn, fn, args);
```

Comment:
Should this use `this`?

```suggestion
      return ReflectApply(fn, this, args);
```

Reply:
yes, and I added tests for this.